packages/expo-calendar/android/src/main/java/expo/modules/calendar/domain/calendar/records/input/NewCalendarInput.kt

```diff
@@ -0,0 +1,58 @@
+package expo.modules.calendar.domain.calendar.records.input
+
+import android.content.ContentValues
+import android.provider.CalendarContract
+import expo.modules.calendar.domain.attendee.enums.AttendeeType
+import expo.modules.calendar.domain.calendar.enums.CalendarAccessLevel
+import expo.modules.calendar.domain.calendar.records.CalendarSource
+import expo.modules.calendar.domain.event.enums.AlarmMethod
+import expo.modules.calendar.domain.event.enums.Availability
+import expo.modules.kotlin.records.Field
+import expo.modules.kotlin.records.Record
+import expo.modules.kotlin.records.Required
+import java.util.TimeZone
+
+data class NewCalendarInput(
+  @Field @Required val name: String,
+  @Field @Required val title: String,
+  @Field @Required val source: CalendarSource,
+  @Field @Required val color: Int,
+  @Field @Required val ownerAccount: String,
+  @Field @Required val accessLevel: CalendarAccessLevel,
+  @Field val isPrimary: Boolean?,
+  @Field val allowedAvailabilities: List<Availability>?,
+  @Field val timeZone: String?,
+  @Field val allowedReminders: List<AlarmMethod>?,
+  @Field val allowedAttendeeTypes: List<AttendeeType>?,
+  @Field val isVisible: Boolean?,
+  @Field val isSynced: Boolean?,
+  @Field val allowsModifications: Boolean?
+) : Record {
+  fun toContentValues() = ContentValues().apply {
+    source.assertValidForNewCalendar()
+
+    put(CalendarContract.Calendars.NAME, name)
+    put(CalendarContract.Calendars.CALENDAR_DISPLAY_NAME, title)
+    put(CalendarContract.Calendars.VISIBLE, isVisible)
+    put(CalendarContract.Calendars.SYNC_EVENTS, isSynced)
+    put(CalendarContract.Calendars.ACCOUNT_NAME, source.name!!)
```

Comment:
Whether `source.name` is null is checked in `source.assertValidForNewCalendar()`
```suggestion
    put(CalendarContract.Calendars.ACCOUNT_NAME, source.name)
```